pkg/store/bucket.go

```diff
@@ -742,15 +738,36 @@ func blockSeries(
 			})
 			s.refs = append(s.refs, meta.Ref)
 		}
+
+		// Reserve chunksLimiter if we save chunks.
 		if len(s.chks) > 0 {
 			if err := chunksLimiter.Reserve(uint64(len(s.chks))); err != nil {
```

Comment:
How about setting hasValidChunk to true in the clause and just checking it in below `if`? Would this work?

Reply:
Done.